weblate/trans/models/component.py

```diff
@@ -882,6 +893,12 @@ def save(self, *args, **kwargs) -> None:
         # the newly created component
         bool(self.source_translation)
 
+        if self.key_filter_re.pattern:
+            with transaction.atomic():
+                translations = Translation.objects.filter(component=self)
+                for translation in translations:
+                    translation.check_sync(force=True)
```

Comment:
This will rescan the files with every component save once key filter is set. I guess the intention was to trigger this once the filter has been changed. Again, this should be done in the `if self.id` block above and change detected using the `old` instance.